asserts/model_test.go

```diff
@@ -195,14 +198,22 @@ func (mods *modelSuite) TestDecodeOK(c *C) {
 		},
 	})
 	// essential snaps included
-	reqSnaps := model.RequiredWithEssentialSnaps()
-	c.Check(reqSnaps, HasLen, len(allSnaps))
-	for i, r := range reqSnaps {
-		c.Check(r.SnapName(), Equals, allSnaps[i].Name)
-		c.Check(r.ID(), Equals, "")
+	reqSnaps := naming.NewSnapSet(model.RequiredWithEssentialSnaps())
+	for _, e := range essentialSnaps {
+		c.Check(reqSnaps.Contains(e), Equals, true)
+	}
+	for _, s := range snaps {
+		c.Check(reqSnaps.Contains(s), Equals, true)
 	}
 	// essential snaps excluded
-	c.Check(model.RequiredNoEssentialSnaps(), DeepEquals, reqSnaps[3:])
+	noEssential := naming.NewSnapSet(model.RequiredNoEssentialSnaps())
+	for _, e := range essentialSnaps {
+		c.Check(noEssential.Contains(e), Equals, false)
+	}
+	for _, s := range snaps {
+		c.Check(noEssential.Contains(s), Equals, true)
+	}
+
```

Comment:
Could we also check that reqSnaps and noEssential were exhausted by above loops? e.g. by simple length checks after the loops (len(reqSnaps) == len(essential)+len(snaps) etc.)?